resources/js/components/shared/TimelinePlot.vue

```diff
@@ -0,0 +1,450 @@
+      // calculate width, each row in the legend looks like: |space-swatch-space-key-space|
+      const legendWidth = 3*legendSpacing + legendRectSize + maxKeyLen;
+      // calculate height: there are numKeys rows, each with height legendRectSize
+      //   and we want to put spacing before each row and also after the last row
+      const legendHeight = numKeys*legendRectSize + (numKeys+1)*legendSpacing;
+
+      // create new SVG group to gather all objects associated with the legend
+      const legend = svg.append("g")
+        .attr('class', 'legend')
+        .attr("width", legendWidth)
+        .attr("height", legendHeight)
+        .attr("transform", `translate(${(width-legendWidth)},0)`);
+      // add SVG rect object to serve as our legend border
+      const legend_box = legend.append("rect")
+        .attr("width", legendWidth)
+        .attr("height", legendHeight)
+        .attr("fill", "white")
+        .attr("stroke", "lightgrey")
+        .attr("opacity", 0.8);
+
+      // dim all lines but the selected one when hovering on a legend item
+      function dimLines(_, i) {
+        for (let j = 0; j < data.length; j++) {
+          if (j !== i) { plot_lines[j].transition().duration(150).attr("opacity", "0.25"); }
+        }
+      }
+      // restore all lines to original opacity on mouse leave
+      function undimLines(_, i) {
+        for (let j = 0; j < data.length; j++) {
+          plot_lines[j].transition().duration(150).attr("opacity", "1");
+        }
+      };
+
+      // add rows to the legend for each dataset
+      const legend_items = legend.selectAll('.legend-item')
+        .data(line_names)
+        .enter()
+        .append('g')
+        .attr('class', 'legend-item')
+        .attr('transform', (d, i) => {
+          // calculate the poisiton of each row with respect to
+          // the origin of the rectangular border we added above
+          let rowHeight = legendRectSize + legendSpacing;
+          let horzDist = legendSpacing;
+          let vertDist = i * rowHeight + legendSpacing;
+          return `translate(${horzDist},${vertDist})`;
+        })
+        .on("mouseover", dimLines)
+        .on("mouseleave", undimLines);
+      // draw the color swatch for each row
+      legend_items.append('rect')
+        .attr('width', legendRectSize)
+        .attr('height', legendRectSize)
+        .style('fill', (d,i) => { return colors[i] });
+      // render the text associated with each dataset
+      legend_items.append('text')
+        .attr('x', legendRectSize + legendSpacing)
+        .attr('y', legendRectSize - legendSpacing/2)
+        .text((d) => { return d })
+        .style("font-size", legend_font_size+"px");
+
+    }
+  }
+}
+</script>
+<style scoped>
+div .timeline-tooltip {
+  position: absolute;
+  text-align: left;
+  opacity: 0;
+  background-color: white;
+  border: solid;
+  border-width: 1px;
+  border-radius: 3px;
+  padding: 5px;
```

Comment:
Unless there's a need for these to be measured in pixels, it would be better to use text-width-based units like `em`.

Reply:
I figured that since they're such small numbers, and since we wouldn't really want the thickness of the tooltip border to change with respect to the page, that it would be ok to use pixels. I can change the padding and radius to `em`, perhaps.